Description: follow flex parameter specification change
 Follow change from *_PARAM to %*-param after deprecation in flex.
 Follow yyerror() parameterisation changes.
Author: Andy Whitcroft <apw@canonical.com>

Index: libsmi-0.4.8+dfsg2/lib/parser-smi.y
===================================================================
--- libsmi-0.4.8+dfsg2.orig/lib/parser-smi.y	2014-04-16 23:49:33.000000000 +0100
+++ libsmi-0.4.8+dfsg2/lib/parser-smi.y	2014-04-17 00:31:50.152357333 +0100
@@ -11,6 +11,9 @@
  * @(#) $Id: parser-smi.y 8090 2008-04-18 12:56:29Z strauss $
  */
 
+%parse-param { struct Parser *parserPtr }
+%lex-param { struct Parser *parserPtr }
+
 %{
 
 #include <config.h>
@@ -43,14 +46,6 @@
 
 
 
-/*
- * These arguments are passed to yyparse() and yylex().
- */
-#define YYPARSE_PARAM parserPtr
-#define YYLEX_PARAM   parserPtr
-
-    
-    
 #define thisParserPtr      ((Parser *)parserPtr)
 #define thisModulePtr     (((Parser *)parserPtr)->modulePtr)
 
Index: libsmi-0.4.8+dfsg2/lib/parser-sming.y
===================================================================
--- libsmi-0.4.8+dfsg2.orig/lib/parser-sming.y	2014-04-16 23:49:33.000000000 +0100
+++ libsmi-0.4.8+dfsg2/lib/parser-sming.y	2014-04-17 00:31:41.288357640 +0100
@@ -11,6 +11,9 @@
  * @(#) $Id: parser-sming.y 7966 2008-03-27 21:25:52Z schoenw $
  */
 
+%parse-param { struct Parser *parserPtr }
+%lex-param { struct Parser *parserPtr }
+
 %{
 
 #include <config.h>
@@ -48,13 +51,6 @@
 #endif
 
 
-/*
- * These arguments are passed to yyparse() and yylex().
- */
-#define YYPARSE_PARAM parserPtr
-#define YYLEX_PARAM   parserPtr
-
-    
     
 #define thisParserPtr      ((Parser *)parserPtr)
 #define thisModulePtr     (((Parser *)parserPtr)->modulePtr)
Index: libsmi-0.4.8+dfsg2/lib/error.h
===================================================================
--- libsmi-0.4.8+dfsg2.orig/lib/error.h	2014-04-17 00:36:14.684348162 +0100
+++ libsmi-0.4.8+dfsg2/lib/error.h	2014-04-17 00:36:31.740347571 +0100
@@ -22,7 +22,7 @@
 #ifdef yyerror
 #undef yyerror
 #endif
-#define yyerror(msg)	smiyyerror(msg, parserPtr)
+#define yyerror(parserPtr, msg)	smiyyerror(msg, parserPtr)
 
 
 extern int smiErrorLevel;	/* Higher levels produce more warnings */
